core/domain/rights_manager.py

```diff
@@ -547,359 +546,252 @@ def _get_activity_rights(activity_type, activity_id):
-        """
-        activity_rights = _get_activity_rights(activity_type, activity_id)
-        if activity_rights is None:
-            return False
-        return self._can_play(activity_rights)
 
-    def can_view(self, activity_type, activity_id):
-        """Returns whether this user has rights to view the given activity.
+def check_can_delete_activity(user_id, user_actions, activity_rights):
+    """Checks whether the user can delete given activity.
 
-        Args:
-            activity_type: str. The type of activity. Possible values:
-                constants.ACTIVITY_TYPE_EXPLORATION
-                constants.ACTIVITY_TYPE_COLLECTION
-            activity_id: str. ID of the activity.
+    Args:
+        user_id: str or None. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        activity_rights: rights_object or None. Rights object of given
+            activity.
 
-        Returns:
-            bool. Whether the user has the right to view the given activity.
-        """
-        return self.can_play(activity_type, activity_id)
+    Returns:
+        bool. Whether the user can delete given exploration.
+    """
+    if activity_rights is None:
+        return False
 
-    # TODO(sll): Add a check here for whether a user is banned or not,
-    # rather than having this check in the controller.
-    def can_edit(self, activity_type, activity_id):
-        """Returns whether this user has rights to edit the given activity.
+    if (activity_rights.is_private() and
+            (role_services.ACTION_DELETE_OWNED_PRIVATE_ACTIVITY in (
+                user_actions)) and
+            activity_rights.is_owner(user_id)):
+        return True
+    elif (activity_rights.is_published() and
+          role_services.ACTION_DELETE_ANY_PUBLIC_ACTIVITY in (
+              user_actions)):
+        return True
+    return False
 
-        Args:
-            activity_type: str. The type of activity. Possible values:
-                constants.ACTIVITY_TYPE_EXPLORATION
-                constants.ACTIVITY_TYPE_COLLECTION
-            activity_id: str. ID of the activity.
 
-        Returns:
-            bool. Whether the user has the right to edit the given activity.
-        """
-        activity_rights = _get_activity_rights(activity_type, activity_id)
-        if activity_rights is None:
-            return False
-        return self._can_edit(activity_rights)
+def check_can_modify_activity_roles(user_id, user_actions, activity_rights):
+    """Checks whether the user can modify roles for given activity.
 
-    def can_delete(self, activity_type, activity_id):
-        """Returns whether this user has rights to delete the given activity.
+    Args:
+        user_id: str or None. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        activity_rights: rights_object or None. Rights Object of given
```

Comment:
changed to " Rights object for the given activity" as it is used elsewhere in the module.